### PR TITLE
Add: Release process for tagging semgrep schema files

### PR DESCRIPTION
**Pull request by ajbt200128**
PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: .github/workflows/release.yml
Comment (spencerdrak) [inline .github/workflows/release.yml:122]: I don't know if this will work without having created a release on that repository prior

File: .github/workflows/release.yml
Comment (spencerdrak) [inline .github/workflows/release.yml:118]: We'll need to configure auth for this repo in one of a few different ways.

File: .github/workflows/release.yml
Comment (spencerdrak) [inline .github/workflows/release.yml:129]: Auth needed (see above)

**Review by spencerdrak — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml
Comment (spencerdrak) [reply .github/workflows/release.yml:122]: Typically a release is created off a tag, so we may need to also create a tag in `returntocorp/semgrep-interfaces`.

**Comment by spencerdrak** [comment]
Looks good overall - few things that need to be addressed through. I am happy to discuss further offline if you'd like @ajbt200128

File: .github/workflows/release.yml
Comment (ajbt200128) [reply .github/workflows/release.yml:122]: Yea looks like we'll have to

File: .github/workflows/release.yml
Comment (ajbt200128) [reply .github/workflows/release.yml:129]: Will the token for this repo not work for the other?

**Review by spencerdrak — COMMENTED** [review]
Approved pending the set up of the authorization bot. Let's work out the details today about setting that up!

**Review by spencerdrak — COMMENTED** [review]
Approved - only comment here is whether or not this has been tested. I don't know how you and the team plan to test this - my strategy has been to set it up in a different repo and run it to ensure I've not missed things, but some sort of test might be useful as well. 

If the team is comfortable moving ahead without a test and live debugging that could also work, but poses some risks of interrupting the release which could get messy.

**Comment by ajbt200128** [comment]
I tested the individual commands/scripts, my main concern is the auth step, but that's one I think we'd have to figure out in prod? I'm not sure how to set up a mock semgrep + interfaces + all the permissions in a way I feel like would be an accurate test at least

**Comment by spencerdrak** [comment]
I set up the `semgrep-ci` bot on test-gh-actions as well as this repo - feel free to test there if you feel necessary, but testing the scripts is a great start. I'll defer to you there. Also, you may want to let @IagoAbal know if this is going to be merged before tomorrow morning - we had some issues with release last week and might want to see if those are ironed out before you merge. Otherwise, nice work!

**Review by spencerdrak — APPROVED** [review]

File: changelog.d/app-2313.added
Comment (IagoAbal) [inline changelog.d/app-2313.added]: I am not sure this should go into the changelog, the changelog is for user-facing changes and this is an internal change.

File: scripts/prepare-schema.py
Comment (IagoAbal) [inline scripts/prepare-schema.py:13]: Why do we need to remove comments?

**Comment by IagoAbal** [comment]
Given that the previous release was not smooth, if this is not urgent, then we may want to wait to see how this release goes. Is that OK? If you think it's urgent, I'm fine merging it now, but I'll revert it if I the release fails and I suspect it may be this.

File: scripts/prepare-schema.py
Comment (ajbt200128) [reply scripts/prepare-schema.py:13]: There's a comment:
```
$comment: "!!If you modify this file, you need to update the submodule in returntocorp/semgrep and returntocorp/semgrep-app!!"
```
which doesn't make sense to me to include as part of the release, so this just removes that

File: changelog.d/app-2313.added
Comment (ajbt200128) [reply changelog.d/app-2313.added]: Removed

File: .github/workflows/start-release.yml
Comment (aryx) [inline .github/workflows/start-release.yml:223]: @ajbt200128 @spencerdrak this broke the release, because we have 2 steps with the same id.
Why do we need an id in the first place anyway?
cc @mjambon

Comment (spencerdrak) [reply .github/workflows/start-release.yml:223]: It's' used for referencing the output of the steps later - I wonder if we can write a semgrep rule to check for duplicate step IDs. If you don't reference the step later, the id can be omitted, although I am in the habit of always adding them.

**Pull request merged**
